OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/19.3.0/setupGridEnv.sh

```diff
@@ -192,22 +192,43 @@ fi
 ####################################### Set Resetting Failed Units ##########################
 resetFailedUnits()
 {
-if [ "${RESET_FAILED_SYSTEMD}" != 'false' ]; then
-
-   cp $SCRIPT_DIR/$RESET_FAILED_UNITS  /var/tmp/$RESET_FAILED_UNITS
-   chmod 755 /var/tmp/$RESET_FAILED_UNITS
-   print_message "Setting Crontab"
-   cmd='su - $GRID_USER -c "sudo crontab $SCRIPT_DIR/$SET_CRONTAB"'
-   eval $cmd
-
-   if [ $?  -eq 0 ];then
-     print_message "Sucessfully installed $SET_CRONTAB using crontab"
-  else
-    error_exit "Error occurred in crontab setup"
-  fi
-
-fi
+        if [ "${RESET_FAILED_SYSTEMD}" != 'false' ]; then
+                # Define the service name
+                SERVICE_NAME="rhnsd"
+
+                # Check if the service is running
+                if pgrep -x "$SERVICE_NAME" >/dev/null; then
+                        print_message "$SERVICE_NAME is running."
+
+                        # Check if the service is responding
+                        if ! systemctl is-active --quiet "$SERVICE_NAME"; then
+                                print_message "$SERVICE_NAME is not responding. Stopping the service."
+                                systemctl stop "$SERVICE_NAME"
+                                systemctl disable "$SERVICE_NAME"
+                                print_message "$SERVICE_NAME stopped."
+                        else
+                                print_message "$SERVICE_NAME is responsive. No action needed."
+                        fi
+                else
+                        print_message "$SERVICE_NAME is not running."
+                fi
+
+        # shellcheck disable=SC2086
+        cp $SCRIPT_DIR/$RESET_FAILED_UNITS  /var/tmp/$RESET_FAILED_UNITS
+        chmod 755 /var/tmp/$RESET_FAILED_UNITS
+        print_message "Setting Crontab"
+        # shellcheck disable=SC2016
+        cmd='su - $GRID_USER -c "sudo crontab $SCRIPT_DIR/$SET_CRONTAB"'
+        # shellcheck disable=SC2086
+        eval $cmd
+        # shellcheck disable=SC2181
+        if [ $?  -eq 0 ];then
```

Comment:
Instead of disabling three different checks, you could combine these lines into a single line that doesn't require disabling any of themm, i.e. 

```shell
if su - "$GRID_USER" -c "sudo crontab $SCRIPT_DIR/$SET_CRONTAB"; then
   print_message "Successfully installed $SET_CRONTAB using crontab"
else
   error_exit "Error occurred in crontab setup"
fi
```